### PR TITLE
Add driver argument to new validation functions

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
When working on https://github.com/NeurodataWithoutBorders/nwbinspector/pull/265 (the integration of the recent validation changes into the NWB Inspector) I ran into an issue getting the cached namespaces for files when using the streaming mode of the Inspector.

Traced it back to these particular function calls - thankfully, `NWBHDF5IO.load_namespaces(...)` already supports a `driver` keyword argument, we just need to add control and propagate it as shown here.

@rly What would your recommendations be for adding tests for this?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
I'll also add the `file` argument propagation in a separate PR for `fsspec` based streaming. That will require just a little bit more since it would need to not conflict with the `io vs. path` usage and just be an alternate usage of the `path`

**Comment by rly** [comment]
This looks good. You can add to https://github.com/NeurodataWithoutBorders/pynwb/tree/dev/tests/integration/ros3 which is the only other place in the tests where we test different drivers.

File: src/pynwb/validate.py [hunk 1]
```diff
@@ -107,6 +109,12 @@ def _get_cached_namespaces_to_validate(path: str) -> Tuple[List[str], BuildManag
         "doc": "Whether or not to print messages to stdout.",
         "default": False,
     },
+    {
+        "name": "driver",
+        "type": list,
```
Comment (bendichter) [inline src/pynwb/validate.py]: shouldn't "driver" be a string, not a list?

**Comment by bendichter** [comment]
This looks like a fix specifically for ros3. It's good to support, but let's try to shift our default over to fsspec for all streaming stuff

File: src/pynwb/validate.py (hunk 1 above)
Comment (CodyCBakerPhD) [inline src/pynwb/validate.py]:
```suggestion
        "type": str,
```

Comment (CodyCBakerPhD) [reply src/pynwb/validate.py]: yes

**Comment by CodyCBakerPhD** [comment]
> This looks like a fix specifically for ros3. It's good to support, but let's try to shift our default over to fsspec for all streaming stuff

https://github.com/NeurodataWithoutBorders/pynwb/pull/1588#issuecomment-1309263552

> I'll also add the file argument propagation in a separate PR for fsspec based streaming. That will require just a little bit more since it would need to not conflict with the io vs. path usage and just be an alternate usage of the path

**Comment by CodyCBakerPhD** [comment]
> This looks good. You can add to https://github.com/NeurodataWithoutBorders/pynwb/tree/dev/tests/integration/ros3 which is the only other place in the tests where we test different drivers.

OK - added tests for the new keyword injections

**Review by rly — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Oh wow, forgot about this one. Thanks @rly

**Pull request merged**